core/io/resource_format_binary.cpp

```diff
@@ -339,7 +339,18 @@ Error ResourceInteractiveLoaderBinary::parse_variant(Variant &r_v) {
 							path = ProjectSettings::get_singleton()->localize_path(res_path.get_base_dir().plus_file(path));
 						}
 
-						RES res = ResourceLoader::load(path, exttype, no_subresource_cache);
+						RES res = nullptr;
+
+						for (List<RES>::Element *E = resource_cache.front(); E; E = E->next()) {
+							if (E->get()->get_path() == path) {
+								res = E->get();
+								break;
+							}
+						}
+
+						if (res == nullptr) {
+							res = ResourceLoader::load(path, exttype, no_subresource_cache);
+						}
```

Comment:
AFAIK `res` can't be `nullptr` technically, it's `Ref<Resource>` so if it's empty/invalid, it will still be a `Ref<Resource>` that passes `res.is_null()`. It just happens to work with `nullptr` as it gets casted implicitly, but it would be cleaner to just do:
``
RES res;

...

if (res.is_null()) {
```
which is consistent with the later check.

Reply:
> AFAIK `res` can't be `nullptr` technically, it's `Ref<Resource>` so if it's empty/invalid, it will still be a `Ref<Resource>` that passes `res.is_null()`. It just happens to work with `nullptr` as it gets casted implicitly, but it would be cleaner to just do:

Fixed. Thanks